### ISSUE TITLE
Support compressed file sources

### ISSUE THREAD

**Issue by joelmarty** [opensearch-project/data-prepper#5245]
**Is your feature request related to a problem? Please describe.**
In order to develop ingestion pipelines for AWS infrastructure component logs (ALB, cloudfront...), I am testing my pipelines with files coming from this infra, which are compressed gzip files.

However, data-prepper does not support compression in the file source plugin, adding an extra step of decompressing the files manually.

In general this is a feature that would be welcomed as compressed log files are a common occurrence.

**Describe the solution you'd like**
the `file` source should have a `compression` field, set to `none` by default. When the plugin is started, the input stream that reads the file comes from the `DecompressionEngine` that corresponds to the `compression` field value.

**Describe alternatives you've considered (Optional)**
N/A

**Additional context**

**Comment by dlvenable** [comment]
Data Prepper already has some code that the `file` source could use for this. The `s3` source is a good working example.

@joelmarty , Would you be able to add this feature to the file source?

**Comment by joelmarty** [comment]
@dlvenable Yes, I was preparing the PR and getting approval from my employer for this contribution. Please review 🙏🏻

**Comment by dlvenable** [comment]
@joelmarty Thank you for the contribution. This should be in the Data Prepper 2.14 release.
